### PR TITLE
[Merged by Bors] - chore(LinearAlgebra): rename to enable `LinearIndependent` dot notation

### PR DESCRIPTION
**Pull request by alreadydone**
+ Rename `cardinal_lift_le_rank_of_linearIndependent`, `cardinal_le_rank_of_linearIndependent(')`, `cardinal_mk/fintype_card/finset_card_le_finrank_of_linearIndependent`, `fintype_card_le_finrank_of_linearIndependent`, `finset_card_le_finrank_of_linearIndependent` by removing trailing `_of_linearIndependent` in favor of namespace `LinearIndependent.`

+ Remove `cardinal_lift_le_rank_of_linearIndependent'`, exact duplicate of the version without the prime

+ Rename `FiniteDimensional/Module.Finite.lt_aleph0_of_linearIndependent` to `LinearIndependent.lt_aleph0_of_finiteDimensional/finite`

+ Add one lemma `LinearIndependent.aleph0_le_rank` in LinearAlgebra/Dimension and two lemmas `LinearIndependent.finrank_eq_zero_of_infinite` and `finrank_eq_nat_card_basis` in LinearAlgebra/Finrank

+ Remove `StrongRankCondition` from `finrank_eq_zero_of_basis_imp_not_finite` and four subsequent lemmas

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by eric-wieser — COMMENTED** [review]
Can you list the renames in the PR description?

**Comment by alreadydone** [comment]
> Can you list the renames in the PR description?

Done! There are actually also three new lemmas.

File: Mathlib/LinearAlgebra/Finrank.lean
Comment (erdOne) [inline Mathlib/LinearAlgebra/Finrank.lean:202]:
```suggestion
theorem finrank_eq_nat_card_basis {ι} [StrongRankCondition K]
    (h : Basis ι K V) : finrank K V = Nat.card ι := by
  rw [Nat.card, ← toNat_lift.{v}, Basis.mk_eq_rank h, toNat_lift, finrank]
```

**Review by erdOne — APPROVED** [review]
LGTM modulo the golf above

**Comment by erdOne** [comment]
I wonder if the `awaiting-author` is a misclick? If so,
maintainer merge

**Comment by jcommelin** [comment]
Please add deprecated aliases for the old lemma names.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**
